### PR TITLE
fix: handle multiple nested includes

### LINKED ISSUE #283: Nested tasks with Includes
### Environment

Device and OS: Ubuntu
App version: 0.5.1 & 0.3.1 & 0.4.1

### Steps to reproduce
Screenshot from 2023-12-15 14-49-50

In the screenshot refer to the cluster-full task definition on line 22. It refers to the cluster-create task definition which refers to an included task from the `setup.yaml` tasks directory. 

When trying to run a `uds run cluster-full` it fails saying that it cannot find the `setup:k3d-cluster task`. @UncleGedd & @zachariahmiller and I all looked at this and the file naming and paths were all correct, so not a naming problem. To further prove that theory, if i were to un-comment the cluster-full definition on line 8 of the screen shot, and run `uds run cluster-full` it would work as expected. 

Also tested this across several uds versions, originally when i encountered this error i was on v0.3.1, tried upgrading to v0.4.1, and then also v0.5.1. Same problem persisted across versions.

### Expected result
Being able to reference an included task in a nested task reference.

### Actual Result
Unable to reference the included task.

### Severity/Priority
Low Priority

### PR DESCRIPTION
**Pull request by decleaver**
## Description

Issue was found where if you reference a task with multiple includes from another task, it errors out not being able to pull in those includes. Not able to process includes of "sibling"

## Related Issue

Fixes #283 

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [ ] [Contributor Guide Steps](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md)(https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md#submitting-a-pull-request) followed

### CODE REVIEW

File: src/test/tasks/tasks.yaml [hunk 1]
```diff
@@ -96,3 +96,6 @@ tasks:
     actions:
       - task: foo:foobar
       - task: remote:echo-var
+  - name: extra-foobar
```
Comment (UncleGedd) [inline src/test/tasks/tasks.yaml:99]: Does any test use this new `extra-foobar` task?

Comment (decleaver) [reply src/test/tasks/tasks.yaml:99]: lol my local debugger, let me create that test

Comment (decleaver) [reply src/test/tasks/tasks.yaml:99]: pushed a test for the rest of the world lol

**Review by UncleGedd — APPROVED** [review]
🚀

**Pull request merged**